operator_rolling_upgrades.md

```diff
@@ -0,0 +1,54 @@
+# Operator Rolling Upgrades
+Optionally allow a Infinispan CR cluster to be upgraded with zero downtime and no loss of data.
+
+## Spec
+Add a field to the CR spec that determines whether an upgrade should be `rolling` or `gracefulShutdown`. The latter is the current approach that does not allow data migration.
+
+```yaml
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: example-infinispan
+spec:
+  upgrades: rolling
+```
+
+## Procedure
+When an upgrade is possible, the operator performs the following for each Infinispan CR:
+
+1. Create a new StatefulSet for the target cluster based upon the CR spec, wait until the cluster is formed
```

Comment:
Is it Ok to create a new `-ping` service for the new Statefulset that utilises a different label value? This will ensure that the two clusters don't merge.

Reply:
I believe this will be necessary. Shouldn't cause any issues.